sw/device/rom_exts/manifest.rs.tpl

```diff
@@ -19,3 +19,31 @@ pub const ${region.size_words_name().as_c_define()}:u32 = ${region.size_words};
 pub const ${offset.offset_name().as_c_define()}:u32 = ${offset.offset};
 
 %endfor
+
+pub struct Field {
+    name: &'static str,
+    offset: u32,
+    size_in_bytes: u32,
+    size_in_words: u32,
+}
+
+const FIELDS: [Field; ${len(regions)}] = [
```

Comment:
I personally would have done `&[Fields]` to avoid having to include the length, but that's just me.

Reply:
I meant writing
```
const FIELDS:  &[Field] = &[...];
```
instead. They're basically equivalent, but it avoids having to include a numeric length (and, as a bonus, makes it harder for people to accidentally depend on that length at build-time).

Reply:
This entirely gone now, I just use named hjson fields and use closure to avoid code duplication, the idea of this was okay when I thought there would be much more field sin the config.